src/Bindings/ManualBindings.cpp

```diff
@@ -367,23 +367,19 @@ static int tolua_StringSplitAndTrim(lua_State * tolua_S)
 /** Retrieves the log message from the first param on the Lua stack.
 Can take either a string or a cCompositeChat.
 */
-static AString GetLogMessage(lua_State * tolua_S)
+static void LogFromLuaStack(lua_State * tolua_S, eLogLevel a_LogLevel)
 {
 	tolua_Error err;
 	if (tolua_isusertype(tolua_S, 1, "cCompositeChat", false, &err))
 	{
-		return static_cast<cCompositeChat *>(tolua_tousertype(tolua_S, 1, nullptr))->ExtractText();
+		auto Msg = static_cast<cCompositeChat *>(tolua_tousertype(tolua_S, 1, nullptr))->ExtractText();
+		Logger::LogSimple(Msg, a_LogLevel);
+		return;
 	}
-	else
-	{
-		size_t len = 0;
-		const char * str = lua_tolstring(tolua_S, 1, &len);
-		if (str != nullptr)
-		{
-			return AString(str, len);
-		}
-	}
-	return "";
+
+	size_t len = 0;
+	const char * str = lua_tolstring(tolua_S, 1, &len);
+	Logger::LogSimple(std::string_view(str, len), a_LogLevel);
```

Comment:
Will lua_tolstring not return nullptr then? There was a check before.

Reply:
If it is `nullptr` then `len` will be 0 so we are okay.

Reply:
I'm not sure, cppreference says

>  The behavior is undefined if [s, s+count) is not a valid range (even though the constructor may not access any of the elements of this range)

better be safe I think

Reply:
`[s, s + count)` becomes `[s, s)` when `count = 0`. which is an empty range, therefore nothing has to be valid.

Reply:
Very well, confirmed by http://www.open-std.org/jtc1/sc22/wg21/docs/papers/2018/p0903r0.pdf